tests/test_image_dataset.py

```diff
@@ -34,15 +34,17 @@ def test_get_data_dynamic(self):
                             load_in_memory=False)
         stream = DataStream(ds, iteration_scheme=ShuffledScheme(
             ds.num_examples, batch_size=10))
-        for imgs, _ in stream.get_epoch_iterator():
+        for imgs, file_paths in stream.get_epoch_iterator():
             assert len(imgs) == 5
             assert imgs[0].shape == (512, 512, 3)
+            assert self.ds_path in file_paths[0]
```

Comment:
Why not `assert all(file_path in dummy_image_paths for image_path in file_paths)`? I understand that you don't want to test whether glob works, but given that there is a custom function `_list_fancy_indexing` in between it wouldn't hurt to actually check the correctness of the output.